homeassistant/components/switchbee/__init__.py

```diff
@@ -0,0 +1,156 @@
+"""The SwitchBee Smart Home integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from switchbee.api import CentralUnitAPI, SwitchBeeError
+from switchbee.device import DeviceType
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    CONF_DEFUALT_ALLOWED,
+    CONF_DEVICES,
+    CONF_SWITCHES_AS_LIGHTS,
+    DOMAIN,
+    SCAN_INTERVAL_SEC,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up SwitchBee Smart Home from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    central_unit = entry.data[CONF_HOST]
+    user = entry.data[CONF_USERNAME]
+    password = entry.data[CONF_PASSWORD]
+    devices_map: dict[str, DeviceType] = {s.display: s for s in DeviceType}
+    allowed_devices = [
+        devices_map[device]
+        for device in entry.options.get(CONF_DEVICES, CONF_DEFUALT_ALLOWED)
+    ]
+    websession = async_get_clientsession(hass, verify_ssl=False)
+    api = CentralUnitAPI(central_unit, user, password, websession)
+    try:
+        await api.connect()
+    except SwitchBeeError:
+        return False
+
+    coordinator = SwitchBeeCoordinator(
+        hass,
+        api,
+        SCAN_INTERVAL_SEC,
+        allowed_devices,
+        entry.data[CONF_SWITCHES_AS_LIGHTS],
+    )
+    await coordinator.async_config_entry_first_refresh()
+    entry.async_on_unload(entry.add_update_listener(update_listener))
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+async def update_listener(hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+    """Update listener."""
+    await hass.config_entries.async_reload(config_entry.entry_id)
+
+
+class SwitchBeeCoordinator(DataUpdateCoordinator):
```

Comment:
Consider moving the coordinator into a separate module (`coordinator.py`), which is the standard.